### ISSUE TITLE
GitHub plugin

### ISSUE THREAD

**Issue by plicease** [Perl5-Alien/Alien-Build#105]
We can use the release page of a github project to select a latest version of a package: `https://github.com/:user/:project/releases`, but this brings in an extra dep (`HTML::LinkExtor`) and requires scraping the page.  This sort of scrape is frequently necessary as many projects do not provide a better way of getting a list of available tarballs, but GitHub has an API, which we can probably use with just `HTTP::Tiny` and `JSON::PP`, both of which are in-core as of recent Perls.

**Comment by plicease** [comment]
I wrote this plugin as an experiment:

https://metacpan.org/pod/Alien::Build::Plugin::Download::GitHub

Unfortunately, I started seeing 403s in cpantesters almost immediately.  Late I was able to reproduce this while fixing some unrelated bugs on my own machine.  Problem is of course that the GitHub API is rate-limited.

https://github.com/Perl5-Alien/Alien-Build-Plugin-Download-GitHub/issues/3

So a GitHub plugin might still be useful, but the API may not be helpful in implementing it.
